central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -150,9 +153,19 @@ func (s *serviceImpl) getProfileStats(ctx context.Context, parsedQuery *v1.Query
 	if err != nil {
 		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results count for request %v", countQuery)
 	}
+	profileMap := map[string]*storage.ComplianceOperatorProfileV2{}
+	for _, scan := range scanResults {
+		profileResults, err := s.profileDS.SearchProfiles(ctx, search.NewQueryBuilder().
+			AddExactMatches(search.ComplianceOperatorProfileName, scan.ProfileName).ProtoQuery())
+		if err != nil || len(profileResults) == 0 {
+			return nil, errors.Wrap(err, "Unable to retrieve compliance profile")
```

Comment:
I think this would not work. If `err == nil` in `errors.Wrap` it will return nil and not wrap the error.

See implementation:

```
// Wrap returns an error annotating err with a stack trace
// at the point Wrap is called, and the supplied message.
// If err is nil, Wrap returns nil.
func Wrap(err error, message string) error {
	if err == nil {
		return nil
	}
	err = &withMessage{
		cause: err,
		msg:   message,
	}
	return &withStack{
		err,
		callers(),
	}
}
```